copier/settings.py

```diff
@@ -35,7 +35,7 @@ def from_file(cls, settings_path: Path | None = None) -> Settings:
             if env_path:
                 settings_path = Path(env_path)
             else:
-                settings_path = user_config_path("copier") / "settings.yml"
+                settings_path = user_config_path(appname="copier", appauthor=False) / "settings.yml"
```

Comment:
Copier users on Windows may already use the `<CONFIG_ROOT>/copier/copier` path. I agree that it should be `<CONFIG_ROOT>/copier` on all platforms, but we should retain backwards compatibility and raise a deprecation warning:

```suggestion
                settings_path = (
                    user_config_path("copier", appauthor=False) / "settings.yml"
                )
                # NOTE: Remove after a sufficiently long deprecation period.
                if OS == "windows":
                    old_settings_path = user_config_path("copier") / "settings.yml"
                    if old_settings_path.is_file():
                        warnings.warn(
                            f"Settings path {old_settings_path} is deprecated. "
                            f"Please migrate to {settings_path}.",
                            DeprecationWarning,
                            stacklevel=2,
                        )
                        settings_path = old_settings_path
```

For this, you need to add the import `from .tools import OS`.

Reply:
@sisp Thanks for the improvement! Since the `copier/copier` path was undocumented and recently introduced with `v9.5.0` I think a rather short deprecation method would suffice.

Forgive my ignorance, this is my first pull request ever :) 
How is future code removal handled, simply through the `# NOTE`? Would it make sence to specify a target version for removal in the comment?

Reply:
I *cannot* miss this occasion to advertise [Yore](https://github.com/pawamoy/yore) :smile: